app/scripts/services/dimStoreService.factory.js

```diff
@@ -803,6 +843,62 @@
 
       return quality;
     }
+    
+    function buildItemMap(stores) {
+      var item_map = {};
+      _.each(stores, function(store, id) {
+        var items = _.each(store.items, function(item) {
+            item_map[item.id] = {name: item.name, type: item.type};
+        });
+      });
+      return item_map;
+    }
+    
+    function clearStaleNewItems(currItems, newItems) {
+      var newItemsClean = {};
+      for(var newItem in newItems) {
+        if (typeof currItems[newItem] === 'undefined') {
```

Comment:
I rarely bother with checks against undefined and the like. JS truthiness is usually more succinct.